lib/game.rb

```diff
@@ -54,8 +55,59 @@ def select_difficulty_level
     ###
   end
 
-  def select_ship_attributes
-    default_attributes = [['Cruiser',3],['Submarine',2]]
+  def select_ship_attributes(board_area, rows, columns)
+
+    @message.would_you_like_to_customize_ships?
+    valid_selections = [1, 2, 3, 4]
+    ship_selection = gets.chomp.to_i
+
+    until valid_selections.include?(ship_selection)
+      @message.invalid_customize_ship_selection
+      ship_selection = gets.chomp.to_i
+    end
+
+    # Ship lengths must be < 1/3 of the board area to play
+    case ship_selection
+    when 1
+      if (board_area / 3) < 5
+        @message.board_is_too_small_for_ship_selection
+        return select_ship_attributes(board_area)
+      end
+      return [['Cruiser', 3],['Submarine', 2]]
+    when 2
+      if (board_area / 3) < 9
+        @message.board_is_too_small_for_ship_selection
+        return select_ship_attributes(board_area)
+      end
+      return [['Battleship', 4],['Cruiser', 3],['Submarine', 2]]
+    when 3
+      if (board_area / 3) < 14
+        @message.board_is_too_small_for_ship_selection
+        return select_ship_attributes(board_area)
+      end
+      return [['Carrier', 5],['Battleship', 4],['Cruiser', 3],['Submarine', 2]]
+    when 4
+      return user_custom_ships(board_area, rows, columns)
+    end
+
+  end
+
+  def user_custom_ships(board_area, rows, columns)
+    user_input = "Y"
+    user_ships = []
+
+    until user_input == "N"
+      user_ships_total_length = user_ships.map { |ship| ship[1] }.inject(0){ |sum, x| sum + x }
```

Comment:
Haven't tried but how about `user_ships.sum{ |ship| ship[1]}`

I know what this line is trying to do but honestly do not understand all of it

Reply:
Yes, that is way simpler.  Updated.